modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/ShapefileFeatureSource.java

```diff
@@ -317,15 +317,6 @@ protected int getCountInternal(Query query) throws IOException {
         SimpleFeatureType readSchema = getReadSchema(q);
         GeometryFactory geometryFactory = getGeometryFactory(q);
 
-        // grab the target bbox, if any
-        Envelope bbox = new ReferencedEnvelope();
-        if (q.getFilter() != null) {
-            bbox = (Envelope) q.getFilter().accept(ExtractBoundsFilterVisitor.BOUNDS_VISITOR, bbox);
-            if(bbox == null) {
-                bbox = new ReferencedEnvelope();
-            }
```

Comment:
Think I am missing the plan, what happened to checking and using the bbox? I just see this code being deleted...